packages/modals/src/styled/StyledModal.ts

```diff
@@ -6,35 +6,105 @@
+import { StyledFooter } from '../styled';
 
-const COMPONENT_ID = 'modals.modal_view';
+const COMPONENT_ID = 'modals.modal';
 
 export interface IStyledModalProps {
   isLarge?: boolean;
+  isCentered?: boolean;
   isAnimated?: boolean;
 }
 
+const animationName = keyframes`
+  0% {
+    transform: scale(0);
+    opacity: 0;
+  }
+
+  50% {
+    transform: scale(1.05);
+  }
+
+  to {
+    opacity: 1;
+  }
+`;
+
+const animationStyles = css`
+  animation-name: ${animationName};
+  animation-duration: 0.3s;
+  animation-timing-function: ease-in-out;
+`;
+
+const boxShadow = (props: ThemeProps<DefaultTheme>) => {
+  const { theme } = props;
+  const { space, shadows } = theme;
+  const offsetY = `${space.base * 5}px`;
+  const blurRadius = `${space.base * 7}px`;
+  const color = getColor('neutralHue', 800, theme, 0.35);
+
+  return shadows.lg(offsetY, blurRadius, color as string);
+};
+
+const largeFooterStyles = () => css`
+  ${StyledFooter} {
+    border: ${props => `${props.theme.borders.sm} ${getColor('neutralHue', 200)}`};
+    padding: ${props => `${props.theme.space.base * 8}px ${props.theme.space.base * 10}px`};
+  }
+`;
+
 /**
- * Accepts all `<div>` props
+ * 1. IE11 centering hack.
  */
-export const StyledModal = styled.div.attrs<IStyledModalProps>(props => ({
+export const StyledModal = styled.div.attrs<IStyledModalProps>({
   'data-garden-id': COMPONENT_ID,
-  'data-garden-version': PACKAGE_VERSION,
-  className: classNames(ModalStyles['c-dialog'], {
-    // Sizing
-    [ModalStyles['c-dialog--large']]: props.isLarge,
-
-    // States
-    [ModalStyles['is-open']]: props.isAnimated,
-
-    // RTL
-    [ModalStyles['is-rtl']]: isRtl(props)
-  })
-}))<IStyledModalProps>`
+  'data-garden-version': PACKAGE_VERSION
+})<IStyledModalProps>`
+  display: flex;
+  position: fixed;
+  flex-direction: column;
+  margin: ${props => (props.isCentered ? '0' : `${props.theme.space.base * 12}px`)};
+  border: ${props => `${props.theme.borders.sm} ${getColor('neutralHue', 300)}`};
+  border-radius: ${props => props.theme.borderRadii.md};
+  border-color: transparent;
+  box-shadow: ${boxShadow};
+  background-color: ${props => props.theme.colors.background};
+  width: ${props => (props.isLarge ? '800px' : '542px')};
+  min-height: 60px;
+  max-height: calc(100vh - 96px);
+  overflow: auto;
+  direction: ${props => props.theme.rtl && 'rtl'};
+
+  &:focus {
+    outline: none;
+  }
+
+  /* stylelint-disable-next-line */
+  @media (max-height: 399px) {
+    top: ${props => props.theme.space.base * 6}px;
+    bottom: auto;
+    margin-bottom: ${props => props.theme.space.base * 6}px;
+    max-height: none;
+  }
+
+  /* stylelint-disable-next-line */
+  @media (max-width: ${props => (props.isLarge ? '799px' : '699px')}) {
+    right: ${props => props.theme.rtl && props.theme.space.base * 6}px;
+    left: ${props => (props.theme.rtl ? 'auto' : `${props.theme.space.base * 6}px`)};
```

Comment:
Use `${props.theme.rtl ? 'right' : 'left}: [calculated-value]` instead of the two properties.